pkg/proxy/winkernel/proxier.go

```diff
@@ -352,6 +353,11 @@ func (proxier *Proxier) newServiceInfo(port *v1.ServicePort, service *v1.Service
 	if err != nil {
 		preserveDIP = false
 	}
+	dsr := service.Spec.ExternalTrafficPolicy == v1.ServiceExternalTrafficPolicyTypeLocal
+	err = hcn.Slash32EndpointPrefixesSupported()
```

Comment:
We need to use DSRSupported() here.

Reply:
@JocelynBerrendonner I am using the Slash32EndpointPrefixesSupported() method here because DSRSupported() does not reflect the backport of the dsr functionality. Some users need this functionality on rs5, which DSRSupported() will inaccurately say is not supported. I can update the version check in hcsshim (assuming the preserveDIP functionality has also been backported) and file a bug to change this version check the next time an hcsshim release is vendored into k8s

Reply:
@elweb9858 : why not do the proper fix in hcsshim today, merge it and pick it up here later today or tomorrow?

Reply:
FWIW, the hcsshim change should be a two lines change in this file: https://github.com/microsoft/hcsshim/blob/27a858bf1651e8204c64b5e6f9979f6c51a0f943/hcn/hcnglobals.go#L41

Reply:
@JocelynBerrendonner you make a fair point, I just want to make sure we are not breaking existing scenarios that use DSRSupported()

Reply:
They shouldn't. Fixing the Hcsshim should actually help everyone. Let's take the conversation offline.

Reply:
Since vendoring in new dependencies does take time, I think it's acceptable to use the version checks we have today (as long as they work properly) and update the shim for future releases.

Reply:
Thanks @elweb9858 ! The latest iteration looks good.